test/keypoints/CMakeLists.txt

```diff
@@ -1,9 +1,23 @@
-PCL_ADD_TEST(keypoints_general test_keypoints
-                 FILES test_keypoints.cpp
-                 LINK_WITH pcl_gtest pcl_io pcl_kdtree pcl_filters pcl_keypoints
-                 ARGUMENTS "${PCL_SOURCE_DIR}/test/cturtle.pcd")
+set(SUBSYS_NAME tests_keypoints)
+set(SUBSYS_DESC "Point cloud library keypoints module unit tests")
+set(SUBSYS_DEPS global_tests keypoints common search kdtree octree features filters)
```

Comment:
I mimic the dependencies from the library/module they belong i.e.: test_keypoints as all the deps the module keypoints has. I don't remember exactly why I added all the recursive dependencies explicitly, but I think it's to ensure the include directories and link flags are properly defined when creating each individual test target.

It's been a while now, my memory is not that fresh :/